### ISSUE TITLE
Blocked after only using registerTool!

### ISSUE THREAD

**Issue by javadevs2016** [mgp25/Chat-API#1930]
Make sure these boxes are checked before submitting your issue -- thank you!

    [x ] You are using PHP > 5.6
    [x] You have installed all extensions and modules listed in here. PHP Protobuf and Curve25519 are required!
    [x] You have checked if someone has already asked the same issue you have.
    [x] You have read the wiki.
    [x] Is not a programming question.
    [x] You are using latest Chat API code.

Is the mere use of Chat-API grounds for blocking? Is this going to get fixed or even recognized as an issue? I got blocked after only using the registerTool! I didn't even get to send a single test message to my phone.

The first try went OK at first, I got my activation code. When I passed it to the tool, I got a failure "_temporary unavailable_". I waited about 30 minutes before trying again and to my surprise, I got blocked.

 [Screenshot here](http://imgur.com/3sfeNSv)

I DID NOT send a single message, so I'd appreciate if you don't call me a spammer or whatever. I've read every related issue at it seems to me that it's a recent problem of unknown origin affecting everyone.

Below is the output of the register tool

**FIRST TRY** 

[examples]# /usr/local/php5.6.18/bin/php registerTool.php
####################
# WA Register Tool      #
####################

Username (country code + number, do not use + or 00):
```

Type sms or voice: Array
(
    [cc] => XX
    [in] => XX
    [lg] => XX
    [lc] => XX
    [id] => _)!▒(-▒冤e$▒▒%h6▒▒
    [token] => sYn81VtIG578rYR0Rur1+hxQfgo=
    [mistyped] => 6
    [network_radio_type] => 1
    [simnum] => 1
    [s] =>
    [copiedrc] => 1
    [hasinrc] => 1
    [rcmatch] => 1
    [pid] => 1041
    [rchash] => 672d92b6d51b1b988e3729d4b0dcbcddc7dc97edc80c08274db397c89df1b0ff
    [anhash] => ec98ce9d5b66f107c11c7e8bd1be67ca
    [extexist] => 1
    [extstate] => 1
    [mcc] => 334
    [mnc] => 000
    [sim_mcc] => 334
    [sim_mnc] => 000
    [method] => sms
)
stdClass Object
(
    [status] => sent
    [length] => 6
    [method] => sms
    [retry_after] => 75
    [sms_wait] => 75
    [voice_wait] => 65
)

```

Enter the received code: 390-620
```
Array
(
    [cc] => XX
    [in] => XX
    [lg] => XX
    [lc] => XX
    [id] => _)!▒(-▒冤e$▒▒%h6▒▒
    [mistyped] => 6
    [network_radio_type] => 1
    [simnum] => 1
    [s] =>
    [copiedrc] => 1
    [hasinrc] => 1
    [rcmatch] => 1
    [pid] => 5427
    [rchash] => bde4a7135fa13ff6809bd404f014a5a694593aa3cfa8426c3683396264276367
    [anhash] => 17b16a69f418f157cde7c06ff24508a6
    [extexist] => 1
    [extstate] => 1
    [code] => 390620
)
stdClass Object
(
    [status] => fail
    [reason] => temporarily_unavailable
    [retry_after] => 3600
)
```

An error occurred registering the registration code from WhatsApp. Reason: **temporarily_unavailable**

**SECOND TRY (30 minutes later)**

[examples]# /usr/local/php5.6.18/bin/php registerTool.php
####################
# WA Register Tool #
####################

Username (country code + number, do not use + or 00): Array
```
(
    [cc] => XX
    [in] => XX
    [lg] => XX
    [lc] => XX
    [id] => _)!▒(-▒冤e$▒▒%h6▒▒
    [mistyped] => 6
    [network_radio_type] => 1
    [simnum] => 1
    [s] =>
    [copiedrc] => 1
    [hasinrc] => 1
    [rcmatch] => 1
    [pid] => 3433
    [extexist] => 1
    [extstate] => 1
)
stdClass Object
(
    [status] => fail
    [reason] => blocked
)
```

**Comment by mgp25** [comment]
If WhatsApp blocks you during registration it means you have registered a lot of numbers under the same network or they just flagged your networks thats why you are blocked instantly.

Funny thing, right?

**Comment by javadevs2016** [comment]
@mgp25 I'm sorry, I just can't see what's so funny about it. The network is fine, otherwise my phone would be blocked as well. This issue is closed, so I won't waste any more of your time and mine with this project.

**Comment by wragnar** [comment]
When you say "under the same network", whatsapp detects only the ip or somethink else from us?

Once you are blocked what can we do not to be recogniced as previously blocked user?

**Comment by Anonim1u** [comment]
Sorry mgp25,
No matter IP,
No matter Network,
Even if you use a single IP , Exclusive, for a single channel registration.
Even if you do this with numerous headers and or different versions.
No matter the use , if idle , if humanized , is automated.
In no more than 48 hours , the recorded channel is blocked.


The Channel Registration , works!
But that identifies that was not generated by the APK.
The locks have time to happen , ever seen.

Tests performed with assurance study rates. :-)

> mgp25

> Using the channel recorded by the tool , even generating tokens WEB , block the channels .

**Comment by Anonim1u** [comment]
Sorry mgp25,
No matter IP,
No matter Network,
Even if you use a single IP , Exclusive, for a single channel registration.
Even if you do this with numerous headers and or different versions.
No matter the use , if idle , if humanized , is automated.
In no more than 48 hours , the recorded channel is blocked.


The Channel Registration , works!
But that identifies that was not generated by the APK.
The locks have time to happen , ever seen.

Tests performed with assurance study rates. :-)

> mgp25

> Using the channel recorded by the tool , even generating tokens WEB , block the channels .

**Comment by Anonim1u** [comment]
if you have something against , please inform that I leave to announce the information.

**Comment by mgp25** [comment]
@Anonim1u im sorry to tell you that you are wrong, because i do a lot of tests very week, with the number, all of them are working and not blocked, i send messages to myself without any issue.

Maybe your tests aren't good enough, the same as i always say, the problem here is that people doesn't do things properly

Regards

**Comment by Anonim1u** [comment]
I reply debug socket , Channels created BY apk and For the tool ; one falls , the other not !
but I respect and I imagine the reason for your answer, your test channel was created BEFORE February .
own channels with over 1 year of use, and all work perfectly , I would answer the same as you ! you are a very capable contact, and I respect , but I believe your answers in this regard , occur for you to use an Old Channel , not again!

**Comment by Anonim1u** [comment]
I can offer help , but I know I have no problem in whatsapp API.
No problem in the API.
The API is PERFECT!
But there is something else in the "register" of channel mode! :-)
Inform it to stop just to stay full of questions on how to use the API, it disturbs!

**Comment by mgp25** [comment]
The 90% of people that does spam refers the numbers to channels, hope you are not of them. As i told you in my previous comment, i do test very week, and that includes registrations, and as i said, it works perfectly. Im working right now with a test number i registered 5 days ago, and i dont have any issue with it.

**Comment by Anonim1u** [comment]
absolutely sure that I am not fam SPAM. to be more direct , use to organize your contacts , receive their e-mail and set up a social profile , facebook , twitter , google + , and others ... not use transmission or sending messages. in my situation I keep the idea that 1 message client = a server message.
nothing other than that.

I have on a server , working channels, but all the channels created using the API before February / March , work perfectly ! I have no problem with them! believe! no problem!

I started trying to see if it could be IP , but I used IP from different locations , exclusively for my use , and the results were blocked channels !

If we create the channel Today , it falls too fast , no matter what to do , regardless of Use;

I have tested many android headers. with all comparative tests ,

I tried to extract the password from Android , which did not succeed ,

Was to generate web access token , simulating CODEQR and providing the same information used header , I had a channel result little more durable , but fell , using or not using.

- Channel 100 % online.
- Channel receiving time after reading , time after editing , pause, sending ..
- Its flows
- Other flows
- I copied the flow of a apk , and I replied with varying times.
- Without any repetition.
- Nothing holds the channel.
- Register your account today , just register , the channel drops. :-)

**Comment by Anonim1u** [comment]
Even check whatsapp library on the channel information , using only the whatsapp has 100 % information.

If you want , I can give you a secure channel , NOT blocks with the API , and Can I Spend a Number To You try Register the channel yourself !

**Comment by beshoo** [comment]
Dear Mr @mgp25 i lost 5 brand new numbers on a brand new VPS server just by registration tool!
as mr @Anonim1u said! exactly same story !

**Comment by decklord** [comment]
@Anonim1u @wragnar @javadevs2016 
It seems that if you try to connect from IP's from cloud providers (GCloud, Amazon, etc) are banned by default by Whatsapp.
Best Regards

**Comment by mpete** [comment]
@Anonim1u @javadevs2016 I'm having the same issue, the channel is blocked immediately after Registration, I tried using different IP and different location with no luck, 1 channel did generate the passwd but was immediately blocked on First Login, I didn't get the chance to Send any message, so I think Whatsapp is blocking registrations coming from API, cause registration from APK is working just fine, please let me know if you find any solution for this, and by the way I only want to recive messages not Send them but I can't even pass the Registration without been blocked.

**Comment by yunopp** [comment]
@Anonim1u @javadevs2016 Have you found any workaround for this issue? Having the same problem?

**Comment by ofirk** [comment]
Same issue here.
Tried everything as Anonim1u mentioned without success.
Any ideas guys?

**Comment by Pereitor** [comment]
I'm suffering from the same issue.

I bought a prepaid SIM/phone number for testing. The only thing I did was trying the registerTool.php example scrit. After getting a WhatsApp registration code via SMS and giving it to the script, it said everything went succesfully. Next attempt was adding it to contacts on my phone and sending a message to it. After that, I was going to try another example script to see if I was able to read the message vis Chat-API, but got plenty of errors. Suspecting I was banned, tried the blockChecker.php script and it confirmed my worries: "Your number is blocked".

I did all testing with my home internet connection, not using any cloud provider, and from localhost. I use this wifi connection for my regular phone and Whatsapp it's still working on it without problems. 

Last attempt was installing official Whatsapp in an Android test mobile phone with my prepaid SIM, but all it says is that I'm not allowed to use Whatsapp with this number.

If this API is not working anymore or if it is not recommended to register via the script, it would be fair to state it somewhere.
